src/components/Input/Input.tsx

```diff
@@ -106,6 +107,7 @@ export const Input = (
     if (onChange) {
```

Comment:
to remove the react-hooks/exhaustive-deps you can change the `setVal(nextVal)` in `setVal(currentValues => ({ ...currentValues, [valuePropName]: nextValue })`

Reply:
nextVal is needed the two lines below anyway:
```
      onChange(event, addonBeforeProp || addonAfterProp ? nextVal : nextValue)
```

Reply:
If you use the state update as it is now you run the risk of running into some bugs related to the asynchronicity of the useState https://react.dev/reference/react/useState#updating-state-based-on-the-previous-Ifate

You could send data to the onChange that is not yet updated.
If you use this approach, you can do everything within the callback function to guarantee the state update.

 ```
 setVal(curentObj => {
   const nextVal = { ...val, [valuePropName]: nextValue }
   if (onChange) {
     onChange(event, addonBeforeProp || addonAfterProp ? nextVal : nextValue)
   }
 })
 ```

Reply:
I set this problem as an issue for the Input component